pages/profile.js

```diff
@@ -3,187 +3,164 @@ import Link from 'next/link'
+      editLink: '#',
+    },
+    {
+      title: 'Phone',
+      fields: ['XXX - XXX - 1234'],
+      editLink: '#',
+    },
+    {
+      title: 'Preferences',
+      fields: ['Give consent to communicate on my behalf'],
+      editLink: '#',
+    },
+  ]
+
+  const [viewCPPFields, setViewCPPFields] = useState(false)
+  const [viewEIFields, setViewEIFields] = useState(false)
+  const [viewOASFields, setViewOASFields] = useState(false)
+
+  const benefitInformations = [
+    {
+      show: viewCPPFields,
+      setShow: setViewCPPFields,
+      fields: fakeFieldsCPP,
+      benefit: t.cpp,
+    },
+    {
+      show: viewEIFields,
+      setShow: setViewEIFields,
+      fields: [...fakeFieldsEI, ...fakeFieldsEI2],
+      benefit: t.ei,
+    },
+    {
+      show: viewOASFields,
+      setShow: setViewOASFields,
+      fields: fakeFieldsOAS,
+      benefit: t.oas,
+    },
+  ]
   return (
     <>
       <h1 className="py-4 text-4xl font-bold text-gray-darker">
-        {t.profileAndSecuritySettings}
+        {t.profileSettings}
       </h1>
-      <TabList
-        titles={[t.profileAndPreferences, t.securitySettings]}
-        onClick={(index) => setTabSelected(index)}
-        tabSelected={tabSelected}
-        containerStyle={'flex flex-wrap border-b border-gray-200'}
-        selectedTabStyle={
-          'bg-gray-100 active border-b-2 border-b-bright-blue-light'
-        }
-        unselectedTabStyle={
-          'text-gray-500 hover:text-gray-600 hover:bg-gray-50'
-        }
-        genericTabStyle={'inline-block text-lg py-3 px-4 text-center'}
-        locale={props.locale}
-      />
-      {tabSelected === 0 ? (
-        <>
-          <TabList
-            titles={[t.ei, t.cpp, t.oas]}
-            onClick={(index) => setProfileTabSelected(index)}
-            tabSelected={profileTabSelected}
-            containerStyle={'my-4 flex flex-wrap gap-4'}
-            selectedTabStyle={'bg-bright-blue-lighter'}
-            unselectedTabStyle={'bg-gray-lighter'}
-            genericTabStyle={
-              'text-center px-10 py-1 whitespace-nowrap rounded-xl hover:bg-bright-blue-lighter '
-            }
-            locale={props.locale}
-          ></TabList>
-          {profileTabSelected === 0 || !profileTabSelected ? (
-            <ProfileInfo
-              fields={[fakeFieldsEI, fakeFieldsEI2]}
-              locale={props.locale}
-            />
-          ) : profileTabSelected === 1 ? (
-            <ProfileInfo fields={[fakeFieldsCPP]} locale={props.locale} />
-          ) : profileTabSelected === 2 ? (
-            <ProfileInfo fields={[fakeFieldsOAS]} locale={props.locale} />
-          ) : null}
-        </>
-      ) : tabSelected === 1 ? (
-        <ProfileInfo fields={[fakeSecurityFields]} locale={props.locale} />
-      ) : null}
+      <hr className="border-1 border-red-400" />
+      <span className="text-lg">{t.updateProfile}</span>
+
+      {benefitInformations.map((benefitInfo) => {
+        return (
+          <>
```

Comment:
I think it might be a good idea to include all of the mapped Info into the ProfileInfo component (including the state change of the view more.

Reply:
Maybe a larger component that maps the benefit information as it does on the page instead? As the component is right now, you could put it on the security page without any changes required.

Reply:
Not sure if that's something that we want at all though.

Reply:
The security page is quite static, so it doesn't quit relate. We can re-visit later if need if you feel this is the better implementation for now.

Reply:
Hmm, yeah sure let's revisit for now.